x/evm/keeper/grpc_query.go

```diff
@@ -754,3 +755,197 @@ func getChainID(ctx sdk.Context, chainID int64) (*big.Int, error) {
 	}
 	return big.NewInt(chainID), nil
 }
+
+// ------------------------------
+// StateDB queries for SGX party
+// ------------------------------
+
+// QueryGetHashStateDB queries hash in statedb for sgx
+func (k Keeper) QueryGetHashStateDB(_ context.Context, req *types.GetHashRequest) (*types.GetHashResponse, error) {
+	ctx := k.preparedCtxs[req.HandlerId]
+	if ctx == nil {
+		return nil, errors.New("context invalid")
```

Comment:
What happens when the validator restarts the node (for whatever reason), either the cosmos one or the SGX one. The preparedCtxs map (or the ctxs map on SGX) will be reset, so then the node will not be able to sync with other nodes. Is that correct?

Reply:
@amaury1093 yeah correct.

Reply:
Shall we store the context in kvstore for persistency?

Reply:
I don't think that's a good idea either. Context has a lot of non-serializable data, like the multistore.

Reply:
@amaury1093 each context will be used just once. no need to keep track of it actually. if a node is down in the middle of a transaction execution, it will result with error. if sgx binary restarts or down in the middle of the transaction execution or queries, it will also result with error while other nodes ( ethermint  + sgx running on another machine ) don't.

Reply:
if either side down, none of the queries and transaction will get succeed. once he is up, they can make the transaction works. for users, they will get transaction execution failure if they choose the issuing node as RPC endpoint.

Reply:
i think the most critical part is the transaction re-play on issuing node for those transactions that are succeed on another node. in this case, we should recover transaction execution and the only solution would keep track the context on both side.

Reply:
As discussed on slack, let's panic in these cases